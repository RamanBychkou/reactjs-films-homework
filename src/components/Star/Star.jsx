import React from 'react';
import shortid from 'shortid';
import style from './Star.scss';

class Star extends React.Component {
  static starQuantity(quantity) {
    let temp = new Array(Math.ceil(quantity)).fill(quantity);
    temp = temp.map(() => {
      const tempEl = (<i className={style.star} key={shortid.generate()} />);
      return tempEl;
    });
    return temp;
  }

  constructor(props) {
    super(props);
    this.state = { rate: 4.8 };
  }

  render() {
    const { rate } = this.state;
    return (
      <div className={style.starContainer}>
        {Star.starQuantity(rate)}
        <p className={style.rate}>{rate}</p>
      </div>
    );
  }
}

export default Star;
