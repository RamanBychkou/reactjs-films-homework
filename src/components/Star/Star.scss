.starContainer {
  display: flex;
  padding-top: 1rem;
}

.starContainer i {
  margin-right: 0.27rem;
  color: #09ade3;
}

.rate {
  border: 1px solid #09ade3;
  border-radius: 4px;
  font-size: 0.7rem;
  width: 1.7rem;
  height: 1.07rem;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 0.1rem;
}

.star {
  width: 1.3rem;
  height: 1.3rem;
  background-image: url(../../pages/MovieDetailsPage/img/star.png);
  background-size: cover;
  background-repeat: no-repeat;
}
