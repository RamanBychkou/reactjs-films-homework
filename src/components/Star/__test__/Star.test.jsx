import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Star from '../index';

test('Star renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Star />);
  const result = renderer.render(<Star />);
  expect(result).toMatchSnapshot();
});

test('Star renders correctly with rate - undefined', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Star rate={undefined} />);
  const result = renderer.render(<Star rate={undefined} />);
  expect(result).toMatchSnapshot();
});

test('Star Quantity', () => {
  const starQuantity = 5;
  const starElementsArray = Star.starQuantity(starQuantity);
  expect(starElementsArray.length).toBe(starQuantity);
});
