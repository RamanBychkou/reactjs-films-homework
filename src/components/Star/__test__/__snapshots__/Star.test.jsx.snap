// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star renders correctly 1`] = `
<div
  className="starContainer"
>
  <p
    className="rate"
  >
    0
  </p>
</div>
`;

exports[`Star renders correctly with rate - undefined 1`] = `
<div
  className="starContainer"
>
  <p
    className="rate"
  >
    0
  </p>
</div>
`;
